pkg/agent/multicast/mcast_controller.go

```diff
@@ -139,52 +152,91 @@ func (c *Controller) checkLastMember(group net.IP) {
 	c.queue.AddAfter(group.String(), igmpMaxResponseTime)
 }
 
-// clearStaleGroups checks the stale groups which have not been updated for mcastGroupTimeout, and then notifies worker
+// clearStaleGroups checks the stale group members which have not been updated for mcastGroupTimeout, and then notifies worker
 // to remove them from groupCache.
 func (c *Controller) clearStaleGroups() {
 	now := time.Now()
 	for _, obj := range c.groupCache.List() {
 		status := obj.(*GroupMemberStatus)
 		status.mutex.RLock()
 		diff := now.Sub(status.lastIGMPReport)
-		status.mutex.RUnlock()
 		if diff > mcastGroupTimeout {
+			// Notify worker to remove the group from groupCache if all members in tt are not updated mcastGroupTimeout.
 			c.queue.Add(status.group.String())
+		} else {
+			// Create a "leave" event for local member if it is not updated for mcastGroupTimeout.
+			for member, lastUpdate := range status.localMembers {
+				if now.Sub(lastUpdate) > mcastGroupTimeout {
+					ifConfig, _ := c.ifaceStore.GetInterfaceByName(member)
```

Comment:
ifConfig, exists := c.ifaceStore.GetInterfaceByName(member) here
use exists to confirm whether ifConfig is nil, I saw in pkg/agent/multicast/mcast_controller.go: 114
_, exist := status.localMembers[e.iface.InterfaceName]
if ifConfig is nil, it will panic.

Reply:
Greate catch, I would update to create an empty InterfaceConfig only setting with InterfaceName to avoid unexpected crash.